### ISSUE TITLE
Different date dicts from system and unix time

### ISSUE THREAD

**Issue by miv391** [godotengine/godot#60327]
### Godot version

v4.0.alpha6.official [e4f0fc50f]

### System information

Windows 10

### Issue description

Getting date dict from system and from unix time gives slightly different results as the date dict returned by `Time.get_date_dict_from_system()` contains `dst:true`.

However, DST doesn't have anything to do with plain dates, so I think that `Time.get_date_dict_from_unix_time()` works as expected and the `dst` field should be removed from the result of `Time.get_date_dict_from_system()`.

### Steps to reproduce

```gdscript
func test():
	print(Time.get_date_dict_from_system())
	print(Time.get_date_dict_from_unix_time(Time.get_unix_time_from_system()))
```

Output:
```
{year:2022, month:4, day:17, weekday:0, dst:true}
{year:2022, month:4, day:17, weekday:0}
```

**Comment by Calinou** [comment]
cc @aaronfranke

**Comment by miv391** [comment]
> don't see why we should purposefully strip this information. What is your use case, are you trying to compare dates?

I don't have an actual use case. I was just playing with the Time functions and noticed that these two functions are behaving differently.

> This is just not true. If it's April 17th at 11:30 PM in standard time, then it's April 18th at 12:30 AM in daylight time.

I don't agree. You cannot have DST information without time zone and time. All time zones do not use DST and different time zones use DST differently. For example, DST began in US 2:00 on 2022-03-13, but in Finland 3:00 on 2022-03-27. So the time zone is required to know whether in some moment of time `dst` is true or false. The time of clock is also needed. If we known that the date is 2022-03-27 and the location is Finland, `dst` may still be either true or false depending on the time.

`Time.get_datetime_dict_from_system()` returns this kind of dict: `{year:2022, month:4, day:18, weekday:1, dst:true, hour:11, minute:5, second:12}`. In this dict the `dst` field is ok as the OS knows my time zone and the status of DST and the dict also includes time (time zone might be nice too). If the programmer wants to know whether current time is using DST, this is the function to be used. But as `Time.get_date_dict_from_system()` returns only the date, DST is not applicable in my opinion.

In your example the datetime has two different dates. However, when `Time.get_date_dict_from_system()` is used, it returns the current date using the current DST information. There are no "April 17th at 11:30 PM in standard time" case for the OS as the DST is used in April (at least in US and Finland).

Most of the time when you are working with plain dates it will just cause problems to add time zones and DST to the mix. Just last week I was fixing bug in my day job caused by using datetime where plain date should have been used.

**Comment by aaronfranke** [comment]
> There are no "April 17th at 11:30 PM in standard time" case for the OS as the DST is used in April (at least in US and Finland).

Just in the US I can already give you two examples where this case does exist: Hawaii and Arizona (outside of the Navajo Nation).

**Comment by aaronfranke** [comment]
https://github.com/godotengine/godot/blob/master/core/os/time.cpp#L343
https://github.com/godotengine/godot/blob/master/drivers/unix/os_unix.cpp#L219
https://github.com/godotengine/godot/blob/master/platform/windows/os_windows.cpp#L279
https://github.com/godotengine/godot/blob/master/core/os/os.h#L210

I don't want to purposefully remove information already calculated by the internal `OS` class. However, looking at the code, it's arbitrary for it to be included here and not in `OS::get_time`. I would be in favor of either removing it from both Time and OS `get_date` or moving it to `get_time` in both Time and OS. It's true that the `dst` flag is mostly useful when you have both the date and time available.

Also note that if it's removed from `get_date` but included in `get_time`, then we'd need another method to check for DST because Time wraps around OS `get_date` and `get_time` for `get_datetime_dict_from_system`. However, I do also see that OS has to call the same methods in both Windows and Unix, so maybe it's worth making `get_datetime` that only needs to call this once, but then we need a new struct...?

**Comment by miv391** [comment]
> > There are no "April 17th at 11:30 PM in standard time" case for the OS as the DST is used in April (at least in US and Finland).
> 
> Just in the US I can already give you two examples where this case does exist: Hawaii and Arizona (outside of the Navajo Nation).

Well, they are in different time zones (or DST regions, I don't know). But current system time of OS cannot be both in DST and standard time.

**Comment by miv391** [comment]
In my opinion, neither `get_date` nor `get_time` need `dst`. Only `get_datetime` really has use for DST information.

> Also, it occurred to me that if you run this code at 23:59:59.999999, you may get a very wrong output.

That's a good point. I agree that OS should have a `get_datetime` function for getting reliable results by making just one call to OS time API.

**Comment by aaronfranke** [comment]
This was fixed by #65509: https://github.com/godotengine/godot/pull/65509/files#diff-dc56afd85a8aee4e77a485e6f6c48cb655229991a82e691370ee46c58edd7a9bR340-R357

DST is no longer included in either the Date or Time methods, only the DateTime methods (as it should be).
